### PR TITLE
fix(release controller): Only update posts if different and can_edit

### PR DESCRIPTION
**Pull request by sasa-tomic**
Also update some related tests

### CODE REVIEW

File: release-controller/mock_discourse.py [hunk 1]
```diff
@@ -2,12 +2,19 @@
...
+    """A mock Discourse client."""
+
     def __init__(self):
+        """Create a new mock client. The actual host needs to be mocked in the test."""
+        self.host = "http://localhost:55555/"
         self.created_topics = []
         self.created_posts = []
         self.api_username = "test"
+        self.api_key = "test_api_key"
+        self.timeout = 10
```
Comment (LittleChimera) [inline release-controller/mock_discourse.py:14]: this seems to not be used

Comment (LittleChimera) [inline release-controller/mock_discourse.py:14]: this seems to not be used

File: release-controller/mock_discourse.py [hunk 2]
```diff
@@ -2,12 +2,19 @@
...
 class DiscourseClientMock(DiscourseClient):
+    """A mock Discourse client."""
+
     def __init__(self):
+        """Create a new mock client. The actual host needs to be mocked in the test."""
+        self.host = "http://localhost:55555/"
         self.created_topics = []
         self.created_posts = []
         self.api_username = "test"
+        self.api_key = "test_api_key"
```
Comment (LittleChimera) [inline release-controller/mock_discourse.py:13]: how is this and host required in the mock now?

File: release-controller/forum.py [hunk 3]
```diff
@@ -82,10 +83,20 @@ def update(self, changelog: Callable[[str], str | None], proposal: Callable[[str
...
+                post_id = created_posts[i]["id"]
+                content_expected = _post_template(
+                    version_name=p.version_name, changelog=p.changelog, proposal=p.proposal
                 )
+                post = self.client.post_by_id(post_id)
+                if post["raw"] == content_expected:
+                    # log the complete URL of the post
+                    logging.info("post up to date: %s", self.post_to_url(post))
+                    continue
+                if post["can_edit"]:
```
Comment (LittleChimera) [inline release-controller/forum.py:95]:
```suggestion
                elif post["can_edit"]:
```

**Review by LittleChimera — APPROVED** [review]

File: release-controller/mock_discourse.py (hunk 1 above)
Comment (sasa-tomic) [reply release-controller/mock_discourse.py:14]: Some tests complained about these missing values, that's why I added this.

```
E       AttributeError: 'DiscourseClientMock' object has no attribute 'api_key'

../python_deps_pydiscourse/site-packages/pydiscourse/client.py:1629: AttributeError
```

Not sure why this suddenly started being required

**Pull request merged**